src/fu-device.h

```diff
@@ -227,6 +233,12 @@ gboolean	 fu_device_detach			(FuDevice	*self,
 							 GError		**error);
 gboolean	 fu_device_reload			(FuDevice	*self,
 							 GError		**error);
+gboolean	 fu_device_prepare			(FuDevice	*self,
+							 FwupdInstallFlags flags,
+							 GError		**error);
+gboolean	 fu_device_cleanup			(FuDevice	*self,
+							 FwupdInstallFlags flags,
+							 GError		**error);
```

Comment:
How do you envision these working compared to the existing composite prepare and composite cleanups?  In tandem?  or separate efforts?

Reply:
They have to work in tandem, as all the plugins get called for each device. The way I see this working is that the "reconnection" code can live in the superclassed FuDevice rather than having to live in the fu-plugin-foo.c file. The latter will still have to open the device using a locker if required anyway.

Reply:
They have to work in tandem, as all the plugins get called for each device. The way I see this working is that the "reconnection" code can live in the superclassed FuDevice rather than having to live in the fu-plugin-foo.c file. The latter will still have to open the device using a locker if required anyway.

Reply:
OK.